File: YrradiaSDL2/YrradiaTheme/Scenes/MainMenuScene/MainMenuScene.cs
namespace YrradiaSDL2.YrradiaTheme.Scenes.MainMenuScene
{
    public class MainMenuScene : Scene
    {
        public MainMenuScene()
        {
            m_gui.AddComponent(new GUIButton("New game", new(0.5f, 0.33f), true));
            m_gui.AddComponent(new GUIButton("Quit", new(0.5f, 0.40f), true));
        }

        protected override void UpdateDerived()
        {
        }

        protected override void RenderDerived()
        {
            _.imageRenderer.DrawImage("DefaultSceneBackground", 0.0f, 0.0f, 1.0f, 1.0f);
            _.imageRenderer.DrawImage("ForradiaLogo", 0.45f, 0.15f, 0.1f, 0.05f);
        }
    }
}

File: src/Theme0/Scenes/MainMenu/MainMenuScene.cpp
/*
 * Copyright 2025 Andreas Åkerberg
 * This code is licensed under MIT license (see LICENSE for details)
 */

#include "MainMenuScene.hpp"
#include "AddMainMenuSceneGUIComponents/AddMainMenuSceneGUIComponents.hpp"
#include "Core/Rendering/Images/ImageRenderer.hpp"

namespace Forradia
{
    void MainMenuScene::InitializeDerived()
    {
        AddMainMenuSceneGUIComponents(GetGUI());
    }

    void MainMenuScene::UpdateDerived()
    {
    }

    void MainMenuScene::RenderDerived() const
    {
        _<ImageRenderer>().DrawImage("DefaultSceneBackground", 0.0f, 0.0f, 1.0f, 1.0f);
        _<ImageRenderer>().DrawImageWithAutoHeight("ForradiaLogo", 0.35f, 0.1f, 0.3f);
    }
}
